pkg/cmd/roachtest/tests/logical_data_replication.go

```diff
@@ -32,6 +32,34 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+type ycsbWorkload struct {
+	// workload is the YCSB workload letter e.g. a, b, ..., f.
+	workloadType string
+	// initRows is the number of records to pre-load into the user table.
+	initRows int
+	// waitDuration is the duration the workload should run for.
+	debugRunDuration time.Duration
+	// initSplits is the count of initial splits before resuming work
+	initSplits int
+}
+
+func (ycsb ycsbWorkload) sourceInitCmd(tenantName string, nodes option.NodeListOption) string {
+	cmd := roachtestutil.NewCommand(`./cockroach workload init ycsb`).
+		MaybeFlag(ycsb.initRows > 0, "insert-count", ycsb.initRows).
+		MaybeFlag(ycsb.initSplits > 0, "splits", 100).
+		Arg("{pgurl%s:%s}", nodes, tenantName)
+	return cmd.String()
+}
+
+func (ycsb ycsbWorkload) sourceRunCmd(tenantName string, nodes option.NodeListOption) string {
+	cmd := roachtestutil.NewCommand(`./cockroach workload run ycsb`).
+		Option("tolerate-errors").
+		Flag("workload", ycsb.workloadType).
+		MaybeFlag(ycsb.debugRunDuration > 0, "duration", ycsb.debugRunDuration).
+		Arg("{pgurl%s:%s}", nodes, tenantName)
+	return cmd.String()
+}
```

Comment:
Let's make sure this workload implements the entire streamingWorkload interface so we can use it elsewhere:

```
type streamingWorkload interface {
	// sourceInitCmd returns a command that will populate the src cluster with data before the
	// replication stream begins
	sourceInitCmd(tenantName string, nodes option.NodeListOption) string

	// sourceRunCmd returns a command that will run a workload
	sourceRunCmd(tenantName string, nodes option.NodeListOption) string

	runDriver(workloadCtx context.Context, c cluster.Cluster, t test.Test, setup *c2cSetup) error
}
```